src/handlers/gestures/gesture.ts

```diff
@@ -263,6 +269,20 @@ export abstract class BaseGesture<
 
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   prepare() {}
+
+  get shouldUseReanimated(): boolean {
+    // use Reanimated when runOnJS isn't set explicitly and all defined callbacks are worklets
+    return (
+      this.config.runOnJS !== true &&
+      this.handlers.isWorklet.reduce((prev, current) => {
+        if (current !== undefined) {
+          return prev && current;
+        } else {
+          return prev;
+        }
+      }, true)
```

Comment:
```ts
!this.handlers.isWorklet.includes(false)
```

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/1862/commits/48cc495c39e73db2a504c0ac5306483c3e74b851.